pkgs/applications/misc/calibre/default.nix

```diff
@@ -27,20 +27,21 @@
 
 mkDerivation rec {
   pname = "calibre";
-  version = "5.30.0";
+  version = "5.31.1";
 
   src = fetchurl {
     url = "https://download.calibre-ebook.com/${version}/${pname}-${version}.tar.xz";
-    sha256 = "058dqqxhc3pl4is1idlnc3pz80k4r681d5aj4a26v9acp8j7zy4f";
+    sha256 = "sha256-3LGEWuHms54ji9GWSyLl8cFWIRBqHY1Jf/CNPJOywrU=";
   };
 
-  # https://sources.debian.org/patches/calibre/5.30.0+dfsg-1
+  # https://sources.debian.org/patches/calibre/5.31.1+dfsg-1
   patches = [
     #  allow for plugin update check, but no calibre version check
     (fetchpatch {
       name = "0001_only_plugin_update.patch";
-      url =
-        "https://sources.debian.org/data/main/c/calibre/${version}%2Bdfsg-1/debian/patches/0001-only-plugin-update.patch";
+      # this patch doesn't change frequently, so point it to a version available on a
+      # stable Debian release to avoid it being removed later
+      url = "https://sources.debian.org/data/main/c/calibre/5.12.0+dfsg-1/debian/patches/0001-only-plugin-update.patch";
```

Comment:
Alternatively, you could fetch it from the original git repo: https://github.com/debian-calibre/calibre/
(obtained via the VCS link on  https://sources.debian.org/src/calibre/5.31.1+dfsg-3/ )